### PR TITLE
add Utils and testing

### PR DESCRIPTION
**Pull request by DJH23**

### CODE REVIEW

File: ironlibrary/src/main/java/com/example/ironlibrary/model/Issue.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
+package com.example.ironlibrary.model;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.Setter;
+
+@Getter
+@Setter
+@AllArgsConstructor
+public class Issue implements InputValidator{
```
Comment (ainafetse22) [inline ironlibrary/src/main/java/com/example/ironlibrary/model/Issue.java]: @NoArgsConstructor

File: ironlibrary/src/main/java/com/example/ironlibrary/model/Utils.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+     * Updates the quantity of an item, ensuring it cannot go below zero.
+     */
+    public static int quantityUpdate(int currentQuantity, int change) {
+        return Math.max(0, currentQuantity + change);
+    }
+
+    /**
+     * Calculates the difference in days between two dates.
+     */
+    public static long dateDifferenceCalculator(String startDate, String endDate) {
```
Comment (ainafetse22) [inline ironlibrary/src/main/java/com/example/ironlibrary/model/Utils.java]: add String date format validation example DATE= "23/03/2024"

File: ironlibrary/src/test/java/com/example/ironlibrary/UtilsTest.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    @Test
+    void testEmailValidator() {
+        Assertions.assertTrue(Utils.emailValidator("example@example.com"));
+        Assertions.assertFalse(Utils.emailValidator("example@"));
+        Assertions.assertFalse(Utils.emailValidator("example.com"));
+        Assertions.assertFalse(Utils.emailValidator(null));
+    }
+
+    @Test
+    void testIsbnValidator() {
```
Comment (ainafetse22) [inline ironlibrary/src/test/java/com/example/ironlibrary/UtilsTest.java:29]: Which is the real format of the ISBN?

File: ironlibrary/src/main/java/com/example/ironlibrary/model/Utils.java [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+        String isbnRegex = "^(?:ISBN(?:-13)?:? )?(?=[0-9X]{10}$|(?=(?:[0-9]+[- ]){3})[- 0-9X]{13}$|97[89][0-9]{10}$|(?=(?:[0-9]+[- ]){4})[- 0-9]{17}$)(?:97[89][- ]?)?[0-9]{1,5}[- ]?[0-9]+[- ]?[0-9]+[- ]?[0-9X]$";
+        Pattern isbnPattern = Pattern.compile(isbnRegex);
+        Matcher matcher = isbnPattern.matcher(isbn);
+        return matcher.matches();
+    }
+
+    /**
+     * Updates the quantity of an item, ensuring it cannot go below zero.
+     */
+    public static int quantityUpdate(int currentQuantity, int change) {
```
Comment (ainafetse22) [inline ironlibrary/src/main/java/com/example/ironlibrary/model/Utils.java]: change the name to addQuantity, if have no sense this way, will always be more than 0. You should validate that the change value is not negative only.

File: ironlibrary/src/test/java/com/example/ironlibrary/UtilsTest.java [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+        Assertions.assertTrue(Utils.isbnValidator("978-3-16-148410-0"));
+        Assertions.assertTrue(Utils.isbnValidator("0-306-40615-2"));
+        Assertions.assertFalse(Utils.isbnValidator("1234567890123"));
+        Assertions.assertFalse(Utils.isbnValidator(null));
+    }
+
+    @Test
+    void testQuantityUpdate() {
```
Comment (ainafetse22) [inline ironlibrary/src/test/java/com/example/ironlibrary/UtilsTest.java:37]: change the test accordingly

File: ironlibrary/src/test/java/com/example/ironlibrary/UtilsTest.java [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+        Assertions.assertEquals(10, Utils.quantityUpdate(8, 2));
+        Assertions.assertEquals(0, Utils.quantityUpdate(5, -10));
+    }
+
+    @Test
+    void testDateDifferenceCalculator() {
```
Comment (ainafetse22) [inline ironlibrary/src/test/java/com/example/ironlibrary/UtilsTest.java:59]: add test for checking date format of each string.

**Review by ainafetse22 — CHANGES_REQUESTED** [review]
Please add all changes before merging

File: ironlibrary/src/main/resources/application.properties [hunk 7]
```diff
@@ -1,8 +1,9 @@
-spring.application.name=ironlibrary
-spring.datasource.url=jdbc:mysql://localhost:3306/library?serverTimezone=UTC
-spring.datasource.username=ironhack
-spring.datasource.password=ironhack2024
-spring.datasource.driver-class-name=com.mysql.cj.jdbc.Driver
 
-spring.jpa.hibernate.ddl-auto=validate
-spring.jpa.show-sql=true
\ No newline at end of file
+
```
Comment (Katenokk) [inline ironlibrary/src/main/resources/application.properties:2]: hay que dejar las properties originales

**Pull request closed without merging**